README.md

```diff
@@ -22,48 +22,53 @@ We regression-test the library against a selection of those (which will change o
 
 ```
-let result = codec.applyDelta(vcdiffDelta, 
-                deltaID,/*any unique identifier of the resulting object there might be*/ 
-                baseID,/*any unique identifier of the object this delta was generated against there might be */
-                false/*is the message base64 encoded*/);
+let result = decoder.applyDelta(vcdiffDelta);
 ```
-
-`applyDelta` could be called as many times as needed. The `VcdiffDecoder` will automatically retain the last delta application result and use it as a base for the next delta application (it will also check whether the uniqie ids of the deltas, if any, match). Thus it allows applying an inifinte sequence of deltas.
+`applyDelta` could be called as many times as needed. The `VcdiffDecoder` will automatically retain the last delta application result and use it as a base for the next delta application (it will also check whether the unique ids of the deltas, if any, match). Thus it allows applying an infinite sequence of deltas.
 
 `result` would be of type `DeltaApplicationResult`. That is a convenience class that allows interpreting the result in various data formats - string, array, etc.
 
+`CheckedVcdiffDecoder` is a flavor of `VcdiffDecoder` that could be used if deltas and objects against which deltas are generated have unique IDs. `CheckedVcdiffDecoder`'s `setBase` and `applyDelta` methods require these IDs and make sure the deltas are applied to the objects they were generated against. E.g.
+
+```
+let result = checkedDecoder.applyDelta(vcdiffDelta, 
+                deltaID,/*any unique identifier of the delta there might be*/ 
+                baseID/*any unique identifier of the object this delta was generated against there might be */);
+```
+
+There are `base64` flavors of `setBase` and `applyDelta` that would accept `base64` encoded input - `setBase64Base` and `applyBase64Delta`. These are convenience methods and they follow the same logic as `setBase` and `applyDelta`.
+
 ## Common Use Cases
 
-### MQTT
+### MQTT with Binary Payload
 
 ```
-var client = mqtt.connect(<url>, <options>);
-var textDecoder = new encoding.TextDecoder();
-var channelDecoder = new DeltaCodec.VcdiffDecoder();
-client.on('message', function (topic, payload) {
-	var message = textDecoder.decode(payload);
-	var data = message;
+const client = mqtt.connect(<url>, <options>);
+const channelName = 'sample-app-mqtt';
+const channelDecoder = new VcdiffDecoder();
+
+client.on('message', (_, payload) => {
+	let data = payload;
+
 	try {
-		if (DeltaCodec.VcdiffDecoder.isDelta(message, true)) {
-			data = channelDecoder.applyDelta(message, undefined, undefined, true).asUtf8String();
+		if (VcdiffDecoder.isDelta(data)) {
+			data = channelDecoder.applyDelta(data).asUint8Array();
 		} else {
-			channelDecoder.setBase(message);
+			channelDecoder.setBase(data);
 		}
 	} catch(e) {
 		/* Delta decoder error */
```

Comment:
Perhaps worth adding console log line to this example, akin to what you have in the SSE one.

The same applies for the other JavaScript files in this PR.